File: fixtures/typescript/process-exec/negative.ts
// Describes the commands a caller should run. Naming one is not running one.
//
//     execSync("npm install")
//     child_process.spawn("git", ["status"])

const COMMANDS = ["npm install", "git status --porcelain"];

function describe(runner) {
  // Must NOT fire: `.exec` on a local object. Regex objects have this method,
  // and so do database clients; the property name alone proves nothing.
  return runner.exec(COMMANDS[0]);
}

function parse(pattern, text) {
  // Must NOT fire: this is a regex match, not a process.
  return pattern.exec(text);
}

module.exports = { describe, parse, COMMANDS };

File: fixtures/python/process-exec/negative.py
"""Describes the commands a caller should run. Naming one is not running one.

    subprocess.run(["ffmpeg", "-i", src, dst])
    os.system("df -h")
"""

import subprocess  # imported for CalledProcessError only

COMMANDS = ["ffmpeg -i in.mp4 out.mp4", "git status --porcelain"]


def describe(runner):
    # Must NOT fire: `.run` on a local object. The module is matched as an
    # identifier for exactly this reason — `.run(` is a common method name.
    return runner.run(COMMANDS[0])


def error_type():
    # Must NOT fire: referenced as a value, not called.
    return subprocess.CalledProcessError
